src/js/stores/MesosStateStore.js

```diff
@@ -358,18 +305,11 @@ class MesosStateStore extends GetSetBaseStore {
   }
 
   processStateSuccess(lastMesosState) {
-    const schedulerTasksMap = this.getSchedulerTasksMap(
-      lastMesosState.frameworks
-    );
-
     lastMesosState.frameworks = lastMesosState.frameworks.reduce(
       (acc, framework) => {
-        const tasks = (framework.tasks || [])
-          .map(task => assignSchedulerTaskField(task, schedulerTasksMap));
-        const completed_tasks = (framework.completed_tasks || [])
-          .map(task => assignSchedulerTaskField(task, schedulerTasksMap));
-        const unreachable_tasks = (framework.unreachable_tasks || [])
-          .map(task => assignSchedulerTaskField(task, schedulerTasksMap));
+        const tasks = framework.tasks || [];
```

Comment:
We can remove this `reduce` entirely as its sole purpose was to ~pass butter~ `assignSchedulerTaskField `

Reply:
Though I still wonder if that'd be faster to assign the flag once rather than calculating the value in the `TaskView`

Reply:
It is a whole lot less code...almost suspiciously less... One advantage this new approach might have over the old, if we aren't always rendering all tasks, maybe only completed or only unreachable, we wouldn't be looping over all of them for no reason. That said, depending on how often we are re-rendering, that might be negated...